### PR TITLE
[YUNIKORN-2617] Update kindest/node to v1.29.2 for Makefile

### PR DESCRIPTION
**Pull request by ryankert01**
### What is this PR for?
as title, follow up of https://github.com/apache/yunikorn-k8shim/pull/833

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2617

### How should this be tested?
Github CI

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: .github/workflows/pre-commit.yml [hunk 1]
```diff
@@ -43,7 +43,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        k8s: [v1.29.1, v1.28.6, v1.27.10, v1.26.13, v1.25.16, v1.24.17]
+        k8s: [v1.29.2, v1.28.7, v1.27.11, v1.26.14, v1.25.16, v1.24.17]
```
Comment (pbacsko) [inline .github/workflows/pre-commit.yml:46]: Nit: consider rename the JIRA ticket (and the PR/commit) to be more accurate, ie. this change also changes the test matrix.

Comment (ryankert01) [reply .github/workflows/pre-commit.yml:46]: Weird, the changes you mentioned seems to be already merge at #833, commit hash also same as #833. Will further investigate that!

**Comment by pbacsko** [comment]
@ryankert01 please do a proper rebase. Now there are multiple commits that are already on master, quite a few changes in 16 files are shown (not sure why though).

**Comment by ryankert01** [comment]
@pbacsko Looks like it's because I forgot to sync my master branch. Thank you for your time.

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request closed without merging**